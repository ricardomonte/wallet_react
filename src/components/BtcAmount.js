import React from 'react';
import Card, { CardPrimaryContent } from "@material/react-card";
import { Headline5, Headline6 } from '@material/react-typography';
import '@material/react-typography/dist/typography.css';
import '@material/react-card/dist/card.css';
import CoinStyle from '../styles/Coins.module.css'
const formatCurrency = require('format-currency')

const BtcAmount = ({btcAmount}) => {
  const amount = formatCurrency(btcAmount,  { format: '%v %c', code: 'BTC', minFraction: 8, maxFraction: 8  })
  return (
    <Card className={`mdc-elevation--z4 ${CoinStyle.content_size}`}>
      <CardPrimaryContent className={CoinStyle.cardContent}><Headline6>Available funds:</Headline6></CardPrimaryContent>
      <CardPrimaryContent className={CoinStyle.cardContent}><Headline5>{amount}</Headline5></CardPrimaryContent>
    </Card>
  )
}

export default BtcAmount;