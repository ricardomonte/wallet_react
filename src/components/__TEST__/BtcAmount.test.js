import React from 'react';
import '@testing-library/jest-dom';
import { render,screen } from '@testing-library/react';
import BtcAmount from '../BtcAmount';

test('Content from BtcAmount matches snapshot', () => {
  const content = render(<BtcAmount btcAmount={'0.2'} />);
  expect(content.container).toMatchSnapshot();
  expect(screen.getByText('Available funds:')).toBeInTheDocument();
});