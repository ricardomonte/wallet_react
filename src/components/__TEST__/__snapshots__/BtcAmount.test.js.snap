// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content from BtcAmount matches snapshot 1`] = `
<div>
  <div
    class="mdc-card mdc-elevation--z4 content_size"
  >
    <div
      class="mdc-card__primary-action cardContent"
    >
      <h6
        class="mdc-typography mdc-typography--headline6"
      >
        Available funds:
      </h6>
    </div>
    <div
      class="mdc-card__primary-action cardContent"
    >
      <h5
        class="mdc-typography mdc-typography--headline5"
      >
        0.20000000 BTC
      </h5>
    </div>
  </div>
</div>
`;
